src/gui/maptools/qgsmaptoolcapture.cpp

```diff
@@ -476,8 +486,8 @@ void QgsMapToolCapture::cadCanvasMoveEvent( QgsMapMouseEvent *e )
   QgsMapToolAdvancedDigitizing::cadCanvasMoveEvent( e );
 
   const QgsPointXY point = e->mapPoint();
-
-  mSnapIndicator->setMatch( e->mapPointMatch() );
+  if ( canvas()->currentLayer() && canvas()->currentLayer()->isSpatial() )
```

Comment:
This is not the correct approach.
This map tool is a base class and is used to capture even if there is no current layer.
This is breaking this scenerio.

Reply:
@3nids Oops, you're right! Sorry about that. I have proposed a fix here #60166 which I think resolves the issue if you wouldn't mind taking a look.